src/models/resource.ts

```diff
@@ -14,6 +16,8 @@ export interface Resource {
     etag: string;
 }
 
+export type ResourceTypes = Resource | ApiClientAuthorization;
```

Comment:
Do `ApiClientAuthorization extends Resource` with a comment explaining why it is an exceptional model. Use Resource everywhere as before instead of ResourceTypes.